### PR TITLE
docs: Add a new guide for result storages

### PR DESCRIPTION
**Pull request by Manish-k723**
### Description

This PR adds documentation on Crawlee's result storage types, specifically the Key-Value Store and Dataset, providing usage examples and file structures for efficient data management.

- Closes: #479  

- [x] CI passed

### CODE REVIEW

**Review by vdusek — CHANGES_REQUESTED** [review]
It should be a guide for Crawlee Python, but your code samples are in JS. 

If you want to continue working on that, update the code samples to Python. Also, please use the same structure as we have in other guides ([docs/guides](https://github.com/apify/crawlee-python/tree/master/docs/guides)) - code samples are in separate files, we use links to API docs, ...

**Comment by Manish-k723** [comment]
Hi @vdusek updated the code as per requirement:
1. Made the guide for crawlee python which missed earlier.
2. updated the code structure as other guide docs have.

Pls let me know, if you see any other issues in this PR.

**Comment by Manish-k723** [comment]
Hi @vdusek pls check this PR and provide your valuable inputs on this, I am keen to work on this and contribute more to the crawlee community.

**Comment by vdusek** [comment]
Hey @Manish-k723, CI checks are not passing.

**Comment by Manish-k723** [comment]
Hi @vdusek resolved the CI errors

File: docs/guides/code/result_storage_kvstore_basic.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import asyncio
+
+from crawlee.storages import KeyValueStore
+
+
+async def main() -> None:
+    store = await KeyValueStore.open()
+    # Store the screenshot in the key-value store.
+    await store.set_value(key='some-key', value={'foo': 'bar'})
```
Comment (vdusek) [inline docs/guides/code/result_storage_kvstore_basic.py:9]: Are you storing screenshot there?

File: docs/guides/result_storage.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+id: result-storage
+title: Result storage
+description: How to store and retrieve data with KeyValueStore and Dataset in Crawlee
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:4]:
```suggestion
description: How to store and retrieve data with key-value store and dataset in Crawlee.
```

File: docs/guides/result_storage.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+import DatasetBasicExample from '!!raw-loader!./code/result_storage_dataset_basic.py';
+
+import KvStoreBasicExample from '!!raw-loader!./code/result_storage_kvstore_basic.py';
+
+import RsPurgeExplicitlyExample from '!!raw-loader!./code/result_storage_purge_explicitly.py';
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:17]:
```suggestion
import DatasetBasicExample from '!!raw-loader!./code/result_storage_dataset_basic.py';
import KvsBasicExample from '!!raw-loader!./code/result_storage_kvs_basic.py';
import RsPurgeExplicitlyExample from '!!raw-loader!./code/result_storage_purge_explicitly.py';
```

File: docs/guides/code/result_storage_dataset_basic.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import asyncio
```
Comment (vdusek) [inline docs/guides/code/result_storage_dataset_basic.py:1]: Please rename the file to `result_storage_kvs_basic.py`.

File: docs/guides/result_storage.mdx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+The following code demonstrates basic operations of key-value stores:
+
+<Tabs groupId="keyvaluestore">
+    <TabItem value="kvstore_basic_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {KvStoreBasicExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:52]: Why tabs if you have there only one item? That does not make sense. See how this is done in the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage).

File: docs/guides/result_storage.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+The following code demonstrates basic operations of the dataset:
+
+<Tabs groupId="dataset">
+    <TabItem value="dataset_basic_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {DatasetBasicExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:81]: Why tabs if you have there only one item? That does not make sense. See how this is done in the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage).

File: docs/guides/result_storage.mdx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Default storages are cleared before the crawler begins, unless specified otherwise. This clearing occurs as soon as we attempt to access any storage (e.g., through `Dataset.open()`) or when using a default storage with one of the helper methods (such as `Dataset.pushData()`, which internally calls `Dataset.open()`). If we do not explicitly interact with storages in our code, the clearing will take place when the crawler's run method is executed. If we need to clear the storages earlier, we can explicitly use the `purgeDefaultStorages()` helper:
+
+<Tabs groupId="cleaningUpStorage">
+    <TabItem value="rs_purge_explicitly_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {RsPurgeExplicitlyExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:93]: Why tabs if you have there only one item? That does not make sense. See how this is done in the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage).

File: docs/guides/result_storage.mdx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    <TabItem value="dataset_basic_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {DatasetBasicExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
+
+## Cleaning up the storages
+
+Default storages are cleared before the crawler begins, unless specified otherwise. This clearing occurs as soon as we attempt to access any storage (e.g., through `Dataset.open()`) or when using a default storage with one of the helper methods (such as `Dataset.pushData()`, which internally calls `Dataset.open()`). If we do not explicitly interact with storages in our code, the clearing will take place when the crawler's run method is executed. If we need to clear the storages earlier, we can explicitly use the `purgeDefaultStorages()` helper:
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:85]: Identifiers are wrong. Probably just blindly copy-pasted from JS...

File: docs/guides/result_storage.mdx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+To see a real-world example of how to get the input from the key-value store, see the [Screenshots](https://crawlee.dev/python/docs/examples/capture-screenshots-using-playwright) example.
+
+## Dataset
+
+Datasets are utilized for storing structured data, where each object contains the same attributes, like products from an online store or listings for real estate. You can visualize a dataset as a table, with each object represented as a row and its attributes as columns. Datasets function as append-only storage, allowing for the addition of new records but preventing modifications or deletions of existing ones.
+
+Every Crawlee project run is linked to a default dataset, which is generally used to store the results specific to that crawler execution. Utilizing this dataset is optional.
+
+In Crawlee, datasets are represented by the Dataset class. To facilitate writing to the default dataset, Crawlee provides the `Dataset.pushData()` function.
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:62]: Identifiers are wrong. Probably just blindly copy-pasted from JS...

File: docs/guides/result_storage.mdx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+The `MemoryStorage` class manages Crawlee's storage. Throughout the crawler's execution, all data is held in memory, and it is simultaneously written to local files within their respective storage type folders.
+
+## KeyValueStore
+
+The key-value store serves the purpose of saving and retrieving data records or files. Each data record is identified by a unique key and linked to a specific MIME content type. Key-value stores are particularly suited for storing web page screenshots, PDFs, or for maintaining the state of crawlers.
+
+Every Crawlee project run is tied to a default key-value store. By convention, the project’s input and output are saved in this default key-value store under the keys `INPUT` and `OUTPUT`, respectively. Typically, both input and output are in JSON format, though other formats are also acceptable.
+
+In Crawlee, the key-value store is represented by the KeyValueStore class. To facilitate easy access to the default key-value store, Crawlee provides the functions `KeyValueStore.getValue()` and `KeyValueStore.setValue()`.
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:33]: Identifiers are wrong. Probably just blindly copy-pasted from JS...

File: docs/guides/result_storage.mdx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+import RsPurgeExplicitlyExample from '!!raw-loader!./code/result_storage_purge_explicitly.py';
+
+This guide explains the available result storage options in Crawlee and how to use them to store and retrieve data.
+
+## Result storage overview
+
+Crawlee offers various types of result storage, each tailored for specific tasks. By default, data is saved to a local disk in the directory specified by the `CRAWLEE_STORAGE_DIR` environment variable. If this variable is not set, Crawlee will automatically use `./storage` within the current working directory.
+
+The `MemoryStorage` class manages Crawlee's storage. Throughout the crawler's execution, all data is held in memory, and it is simultaneously written to local files within their respective storage type folders.
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:25]: Missing API docs link (all classes should be linked to the API docs).

File: docs/guides/result_storage.mdx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+<Tabs groupId="cleaningUpStorage">
+    <TabItem value="rs_purge_explicitly_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {RsPurgeExplicitlyExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
+
+Invoking this function will clear the default results storage directories, with the exception of the `INPUT` key in the default key-value store directory. This serves as a shortcut to execute the optional `purge` method from the `StorageClient` interface; in other words, it calls the `purge` method of the underlying storage implementation in use. Additionally, this method ensures that the storage is purged only once for a specific execution context, making it safe to call multiple times.
```
Comment (vdusek) [inline docs/guides/result_storage.mdx:95]: Missing API docs link (all classes should be linked to the API docs).

**Review by vdusek — CHANGES_REQUESTED** [review]

File: docs/guides/result_storage.mdx (hunk 5 above)
Comment (vdusek) [reply docs/guides/result_storage.mdx]: I mean, could you please add other tabs as we have for the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage)? Users should find out here how to use the result storages with crawlers.

- Basic usage
- Usage with Crawler
- Explicit usage with Crawler

File: docs/guides/result_storage.mdx (hunk 6 above)
Comment (vdusek) [reply docs/guides/result_storage.mdx]: Same as for KVS.

**Comment by Manish-k723** [comment]
Hey @vdusek updated the PR. thanks

**Review by vdusek — APPROVED** [review]
Thank you @Manish-k723, I did a little more "text polishment" and improved the code examples. Good job!

**Pull request merged**